docs/api.md

```diff
@@ -118,8 +121,46 @@ type AddonsSpec struct {
 }
 
 // TODO maybe we have profiles for scaling behaviors
-type AutoScalerAddonSpec struct {
-    ResourceLimits LimitRange
+type ClusterAutoscaling struct {
+	// Maximum number of nodes in all node groups.
+	// Cluster autoscaler will not grow the cluster beyond this number.
+	// +kubebuilder:validation:Minimum=0
+	MaxNodesTotal int32 `json:"maxNodesTotal"`
+
+	// Should CA scale down the cluster
+    // default: true
+	ScaleDownEnabled bool `json:"enabled,omitempty"`
+
+	// Gives pods graceful termination time before scaling down
+    // default: 600 seconds
+	MaxPodGracePeriod *int32 `json:"maxPodGracePeriod,omitempty"`
+
+	// Maximum time CA waits for node to be provisioned
+    // default: 15 minutes
+	// +kubebuilder:validation:Pattern=^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+	MaxNodeProvisionTime string `json:"maxNodeProvisionTime,omitempty"`
+
+	// To allow users to schedule "best-effort" pods, which shouldn't trigger
+	// Cluster Autoscaler actions, but only run when there are spare resources available,
+    // default: -10
+	// More info: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-work-with-pod-priority-and-preemption
+	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
+
+	// BalanceSimilarNodeGroups enables/disables the
+	// `--balance-similar-node-groups` cluster-autocaler feature.
+	// This feature will automatically identify node groups with
+	// the same instance type and the same set of labels and try
+	// to keep the respective sizes of those node groups balanced.
+    // default: false
+	BalanceSimilarNodeGroups *bool `json:"balanceSimilarNodeGroups,omitempty"`
```

Comment:
I propose removing this one and splitting out a separate discussion about something like a "balancing policy" instead of a boolean

Reply:
i'd like to hear more about the policy enum/union idea. that said, i'm fine to remove these for now.